rio_alpha/alpha.py

```diff
@@ -63,15 +65,19 @@ def _alpha_worker(open_file, window, ij, g_args):
     rgb = open_file[0].read(window=window)
     depth, rows, cols = rgb.shape
 
-    alpha = calc_alpha(rgb,
-                       g_args['ndv'])
+    if g_args['ndv_masks']:
+        mask = open_file[0].read_masks(window=window)
```

Comment:
The intention of `ndv-masks` is to determine the alpha band based on internal GDAL metadata, correct?  The CLI help text mentions "internal masks" but that's only one of the possible cases: see #9. It's not clear that this logic would handle all of these cases properly.

We should consider using Rasterio's  `src.dataset_masks()` which abstracts all of the nodata complexities into a 2D representation suitable as an alpha band.

So we could replace this block with something like:

```
if g_args['ndv_masks']:
    # Regardless of how nodata is represented internally, we get back a 2D, uint8 array
    # 0 = nodata, 255 = valid data
    mask =  open_file[0].dataset_mask(window=window)

    # scale to rgb.dtype
    calc_alpha(mask, [0])
```

@sgillies has a related ticket in Rasterio (https://github.com/mapbox/rasterio/issues/879) which exposes more details about GDAL's internal model. But what I'm advocating here instead is using the existing `dataset_mask` method to *hide* those implementation details with a well-tested heuristic (see the logic at https://github.com/mapbox/rasterio/blob/master/rasterio/_io.pyx#L1151-L1176).